charts/netdata/templates/ingress.yaml

```diff
@@ -26,8 +27,18 @@ spec:
       http:
         paths:
           - path: {{ $ingressPath }}
+            {{- if and (ne $apiVersion "extensions/v1beta1") (ne $apiVersion "networking.k8s.io/v1beta1") }}
```

Comment:
```suggestion
            {{- if not (eq $apiVersion "extensions/v1beta1" "networking.k8s.io/v1beta1") }}
```

> eq (only) accepts two or more arguments and compares the second and subsequent to the first, returning in effect

Reply:
I hope the syntax is correct, didn't test. But you get the idea.

Reply:
No, unfortunately it doesn't work for me correctly. I'm afraid it has to be like that.

My intention was to make this Ingress resource compatible with new versions from the future. I can't just make simple condition `eq "networking.k8s.io/v1"` because future version most likely will use this field too. Then I need to make negative condition twice for `"extensions/v1beta1"` and `"networking.k8s.io/v1beta1"`.

Go Templaces syntax is a little inconvenient but at least works.

Reply:
Sorry, what didn't work? What I did is simplified the expressing using `eq` function feature

> For simpler multi-way equality tests, eq (only) accepts two or more arguments and compares the second and subsequent to the first, returning in effect
> 
> arg1==arg2 || arg1==arg3 || arg1==arg4 ...
>
> (Unlike with || in Go, however, eq is a function call and all the arguments will be evaluated.)

=>


`and (ne $apiVersion "extensions/v1beta1") (ne $apiVersion "networking.k8s.io/v1beta1")`

same as

`not (eq $apiVersion "extensions/v1beta1" "networking.k8s.io/v1beta1")`

which you read as `if apiVersion is not any of ...`

What am i missing?

Reply:
I was missleaded by official documentation for Helm. https://helm.sh/docs/chart_template_guide/function_list/#string-functions doesn't show anything about more than 2 args for `eq`.

Variant with `{{- if not (eq $apiVersion "extensions/v1beta1" "networking.k8s.io/v1beta1") }}` works indeed.

Reply:
Oh, `eq` is [templates built-in](https://golang.org/pkg/text/template/). Helm extends available functions by using http://masterminds.github.io/sprig/